crates/sui/src/unit_tests/cli_tests.rs

```diff
@@ -767,6 +767,47 @@ async fn test_package_publish_command_failure_invalid() -> Result<(), anyhow::Er
     Ok(())
 }
 
+#[sim_test]
+async fn test_package_publish_nonexistent_dependency() -> Result<(), anyhow::Error> {
+    let mut test_cluster = TestClusterBuilder::new().build().await?;
+    let address = test_cluster.get_address_0();
+    let context = &mut test_cluster.wallet;
+
+    let client = context.get_client().await?;
+    let object_refs = client
+        .read_api()
+        .get_owned_objects(
+            address,
+            Some(SuiObjectDataOptions::full_content()),
```

Comment:
nit: Do you really need to get the full contents here?  Seems like you only need the ID.